### DISCUSSION TITLE
Cannot light up SK9822 leds with Adafruit ItsyBitsy RP2040!

### DISCUSSION THREAD

**Discussion by guyisme** [awawa-dev/HyperSerialPico#27] [Q&A]
I'm running HyperHDR v20 from Raspberry Pi 4 with the following hardware:

- MS2130 USB3 capture card (not old version).
- Adafruit ItsyBitsy RP2040 led driver.
- SK9822 RGB led strip (total 306 leds).
- 5V DC 30A psu.

SK9822 have 4 pin connector (GND, Clock, Data, 5V). ItsyBitsy is connected to Raspberry pi through USB2. From ItsyBitsy to SK9822, I connect ground (G), data (GPIO14) and clock (SCL/GPIO3).

From HyperHDR overview panel I can see the capture card and led device is working however I cannot get any light from the leds. I changed the stripe with a one meter one to test but got the same failure. Previously I was running HyperHDR v17 with raspberry pi driving the leds directly. Although not stable I was getting the leds working most of the time. Now with new version and RP2040 I cannot make it work. What could I do wrong, any ideas?

![Ekran görüntüsü 2024-06-21 190842](https://github.com/awawa-dev/HyperSerialPico/assets/114879039/4677459d-5907-4f12-b1a0-877ebfa2a082)

**Comment by awawa-dev** [comment]
Hi
The Adafruit ItsyBitsy only has one output channel using 5V and is not suitable for SPI LEDs. If you use the data line at 5V and the clock line is 3.3V, then the result is unpredictable. You can try to use both 3.3V lines for clock & data but it still doesn't guarantee success since you had the problem on the RPi. And verify your adalight settings and logs, if the rp2040 is detected correctly because the screenshot doesnt tell anything about it.

**Reply by awawa-dev** [reply]
> From ItsyBitsy to SK9822, I connect ground (G), data (GPIO14) and clock (SCL/GPIO3)

And where did you a firmware for such a pinout?

**Reply by guyisme** [reply]
Thank you very much for your quick reply. You can see the copy of my logs here: [https://justpaste.it/f1ujg](url)

I flashed the firmware HyperSerialPico_spi.uf2 and make guess work from [https://learn.adafruit.com/adafruit-itsybitsy-rp2040/pinouts](url)

I understand that If I change the led strip to SK6812 RGBWW the problem would be solved. Or I would use Pimeroni Plasma 2050 or Adafruit Feather 2040 Scorpio with SK9822, am I correct?

**Reply by awawa-dev** [reply]
If you upload the ready-made default standard HyperSerialPico firmware from the release section without any changes, then as in the instructions on the home page, the pinout is as follows and there is no point in looking for it on other projects:
![obraz](https://github.com/awawa-dev/HyperSerialPico/assets/69086569/8c5f0cc3-7867-419b-a3a4-0c4337a62cef)

**Reply by guyisme** [reply]
I tried but unfortunately no luck. The leds light up but no change of colors, effect, etc. It looks like driver cannot control the leds.

**Reply by awawa-dev** [reply]
So if you are sure that you have connected everything correctly (taking into account the common ground between the LED strip and the pico board) then you must use eg. Adafruit Feather 2040 Scorpio or add a level shifter. You can still use some tricks to make the +5v cable to the LED strip much longer than the signal cables (which should never be too long anyway), hoping that the voltage drop will cause the data to be recognized, but this is a rather dodgy solution.
BTW you don't need Force ESP/Pico Detection : ON, the Pico board is detected correctly.
